inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/fetcher/kafka/KafkaMultiTopicsFetcher.java

```diff
@@ -0,0 +1,388 @@
+                || offsetResetStrategy == SortClientConfig.ConsumeStrategy.earliest_absolutely) {
+            properties.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        } else {
+            properties.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "none");
+        }
+        properties.put(ConsumerConfig.FETCH_MAX_BYTES_CONFIG,
+                context.getConfig().getKafkaFetchSizeBytes());
+        properties.put(ConsumerConfig.FETCH_MAX_WAIT_MS_CONFIG,
+                context.getConfig().getKafkaFetchWaitMs());
+        properties.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, false);
+        properties.put(ConsumerConfig.PARTITION_ASSIGNMENT_STRATEGY_CONFIG,
+                RangeAssignor.class.getName());
+        properties.put(ConsumerConfig.CONNECTIONS_MAX_IDLE_MS_CONFIG, 120000L);
+        LOGGER.info("start to create kafka consumer:{}", properties);
+        return new KafkaConsumer<>(properties);
+    }
+
+    @Override
+    public void ack(String msgOffset) throws Exception {
+        // the format of multi topic kafka fetcher msg offset is topic:partitionId:offset, such as topic1:20:1746839
+        String[] offset = msgOffset.split(":");
+        if (offset.length != 3) {
+            throw new Exception("offset is illegal, the correct format is topic:partitionId:offset, "
+                    + "the error offset is:" + msgOffset);
+        }
+
+        // parse topic partition offset
+        TopicPartition topicPartition = new TopicPartition(offset[0], Integer.parseInt(offset[1]));
+        long ackOffset = Long.parseLong(offset[2]);
+
+        // ack
+        if (!ackOffsetMap.containsKey(topicPartition) || !ackOffsetMap.get(topicPartition).containsKey(ackOffset)) {
+            LOGGER.warn("did not find offsetMap or ack offset of {}, offset {}, just ignore it",
+                    topicPartition, ackOffset);
+            return;
+        }
+
+        // mark this offset has been ack.
+        ConcurrentSkipListMap<Long, Boolean> tpOffsetMap = ackOffsetMap.get(topicPartition);
+        // to prevent race condition in AckOffsetOnRebalance::onPartitionsRevoked
+        if (Objects.nonNull(tpOffsetMap)) {
+            tpOffsetMap.put(ackOffset, true);
+        }
+    }
+
+    @Override
+    public void pause() {
+        consumer.pause(consumer.assignment());
+    }
+
+    @Override
+    public void resume() {
+        consumer.resume(consumer.assignment());
+    }
+
+    @Override
+    public boolean close() {
+        this.closed = true;
+        try {
+            if (fetchThread != null) {
+                fetchThread.interrupt();
+            }
+            if (consumer != null) {
+                prepareCommit();
+                consumer.commitSync(commitOffsetMap);
+                consumer.close();
+            }
+            commitOffsetMap.clear();
+        } catch (Throwable t) {
+            LOGGER.warn(t.getMessage(), t);
```

Comment:
No need getMessage when printing all stack traces of t.

Reply:
fixed, thx